client/layout/index.jsx

```diff
@@ -223,5 +228,8 @@ export default connect( state => {
 		authorization, it would remove the newly connected site that has been fetched separately.
 		See https://github.com/Automattic/wp-calypso/pull/31277 for more details. */
 		shouldQueryAllSites: currentRoute && currentRoute !== '/jetpack/connect/authorize',
+		communityTranslatorEnabled:
+			config.isEnabled( 'i18n/community-translator' ) &&
+			getUserSetting( state, ENABLE_TRANSLATOR_KEY ),
```

Comment:
Can we help make the `Layout` component more clean and generic by moving all the community translator logic out of it? The async loading could be done like this:

In `Layout`, we ideally do just:
```jsx
<CommunityTranslator />
```

Then `CommunityTranslator` checks all the conditions and optionally async loads larger subcomponents:
```jsx
function CommunityTranslator( { isEnabled } ) {
  if ( isEnabled && someConditionHolds ) {
    return <AsyncLoad require="translator/v1/launcher" />;
  } else if ( anotherConditionHolds ) {
    return <AsyncLoad require="translator/v2/new-shiny-impl" />;
  } else {
    return null;
  }
}

connect( getUserSettingsEtcFromRedux )( CommunityTranslator );
```